inc/fasta/fasta.hpp

```diff
@@ -34,8 +33,28 @@ namespace ebi
       class IFasta
       {
       public:
+        /**
+         * Extract n base pairs from the contig starting from an offset
+         * @param contig - the name of the contig to extract the sequence from
+         * @param pos - the starting offset position
+         * @param length - the number of base pairs to be extracted
+         * @return the sequence string, empty if nothing can be extracted.
+         */
         virtual std::string sequence(const std::string& contig, const size_t pos, const size_t length) = 0;
-        virtual size_t count(const std::string& contig) const = 0;
+
+        /**
+         * Get the number of times a contig appears in a FASTA.
+         * @param contig - the name of the contig
+         * @return the number of times a contig appears in a FASTA. 0 if the contig is not found.
+         */
+        virtual size_t count(const std::string &contig) const = 0;
```

Comment:
does it really make sense to call this `count`? having the other method for the sequence length, this becomes more and more obscure. I would call it `bool is_present` or `bool is_contig_present`. I guess this is called `count` after bioio::FastaIndex::count which is actually std::unordered_map::count which can only return 0 or 1.

Reply:
And if it can return a greater number, then `sequence_num_occurrences` would still be clearer. I suggest the `sequence` prefix for consistency with the rest of methods or suggestions about them.